lib/GenomeFileUtil/core/FastaGFFToGenome.py

```diff
@@ -530,8 +536,15 @@ def _add_missing_parents(self,feature_list):
                     if("CDS" in ftrs[i]["type"]):
                         new_rna_ftr = copy.deepcopy(ftrs[i])
                         new_rna_ftr["type"] = "mRNA"
-                        new_ftrs.append(new_rna_ftr)
                         ftrs[i]["Parent"]=new_rna_ftr["ID"]
+                        new_ftrs.append(new_rna_ftr)
+
+                #Special case where, if a CDS is annotated to have a Parent, but the Parent is a gene type
+                if("Parent" in ftrs[i] and "CDS" in ftrs[i]["type"] and ftrs[i]["Parent"] in gene_ids_dict):
```

Comment:
Another python trick for you. You can write this line as if `"CDS" in ftrs[i]["type"] and ftrs[i].get("Parent", "") in gene_ids_dict:` The second argument in get is the default value that will be returned if a key is not in the dict.